packages/hawtio/src/preferences/HomePreferences.tsx

```diff
@@ -1,13 +1,46 @@
 import { Button, CardBody, Form, FormGroup, FormSection, Switch } from '@patternfly/react-core'
 import React, { useState } from 'react'
-import { log } from './globals'
+
+type LocalStorageFieldData<T> = {
+  localStorageKey: string
+  defaultValue: T
+  updateFunction: React.Dispatch<React.SetStateAction<T>>
+}
 
 export const HomePreferences: React.FunctionComponent = () => {
-  const [defaultVerticalNavState, setDefaultVerticalNavState] = useState(true)
+  const LOCAL_STORAGE_VERTICAL_NAVIGATION = 'local_storage_vertical_navigation'
+  const DEFAULT_VALUE_VERTICAL_NAVIGATION = true
+
+  const VERTICAL_NAVIGATION_INITIAL_VALUE: boolean =
+    localStorage.getItem(LOCAL_STORAGE_VERTICAL_NAVIGATION) !== null
+      ? localStorage.getItem(LOCAL_STORAGE_VERTICAL_NAVIGATION) === 'true'
+      : DEFAULT_VALUE_VERTICAL_NAVIGATION
+
+  const [defaultVerticalNavState, setDefaultVerticalNavState] = useState(VERTICAL_NAVIGATION_INITIAL_VALUE)
+
+  const VERTICAL_NAVIGATION_FIELD_PARAMETERS: LocalStorageFieldData<boolean> = {
+    localStorageKey: LOCAL_STORAGE_VERTICAL_NAVIGATION,
+    defaultValue: true,
+    updateFunction: setDefaultVerticalNavState,
+  }
+
+  const FIELDS_TO_RESET = [VERTICAL_NAVIGATION_FIELD_PARAMETERS]
 
   const reset = () => {
-    // TODO: impl
-    log.info('TODO - Reset settings')
+    FIELDS_TO_RESET.forEach(field => {
```

Comment:
why can't we just use `localStorage.clear()`?

It's always recommended to consult with the legacy impl to see if we might miss anything.
https://github.com/hawtio/hawtio-core/blob/main/src/app/preferences/reset-preferences/reset-preferences.controller.ts#L11-L29

Reply:
Clear would remove all the settings, moreover, everything on the localstorage, not just the one on that page of the settings, so its generally recommended to avoid it.

Ill check it tomorrow anyway and show a gif too if it is as I think

Reply:
Yes, clear would remove all the settings for the same domain e.g. `http://localhost:3000` but not others. Since the `Reset` button is supposed to reset all the Hawtio settings, it should be what is expected to do. The old Hawtio actually has been with the behaviour. It's kind of hard reset. Do you still think we need to avoid this?
https://github.com/hawtio/hawtio-core/blob/main/src/app/preferences/reset-preferences/reset-preferences.controller.ts#L11-L29

When you look at the legacy impl, there are two things to consider:

First one is which keys should be reserved. Looking at the legacy impl, you'll see `jvmConnect` and `osAuthCreds` are kept. If we follow the same behaviour we should at least reserve the connection settings, i.e. `connect.connections`.

Second one is that we'd need to reload the page once you reset the local storage, since the operation should also clear Jolokia settings which require reload to apply the new settings.

Reply:
Looking at the old implementation I think you're right that there is another way to do this with a preferences service for example. Let me think it through.

Also, yep, you are correct about resetting and I was wondering about that the other day.

I guess I'm going to rewrite it in a more simple way and taking into account all of this